### PR TITLE
Migrate NotificationGateway frontend to React

### PR DESCRIPTION
**Pull request by jakhog**
The frontend has been migrated to React, to align with the rest of the BCs. Also I have added some UI to register new SMS Gateways.

It looks even worse than the last one, but it should now be easier to style it like the other frontends, and someone with taste to contribute 🙂

### CODE REVIEW

**Comment by jakhog** [comment]
Any chance to get some reviews here @woksin / @techgeek03 ?

**Comment by techgeek03** [comment]
Sure I will take a look tomorrow :)

File: Source/NotificationGateway/Web/Dockerfile [hunk 1]
```diff
@@ -7,12 +8,10 @@ RUN yarn install
...
 WORKDIR /CBS/Source/NotificationGateway/Web
-RUN mkdir -p /CBS/Source/NotificationGateway/Core/wwwroot
-RUN npm run build
+RUN yarn build
 
 # Build runtime image
 FROM nginx:1.15-alpine
-COPY --from=build /CBS/Source/NotificationGateway/Core/wwwroot /usr/share/nginx/html
+COPY --from=build /CBS/Source/NotificationGateway/Web/dist /usr/share/nginx/html
 COPY --from=build /CBS/Source/NotificationGateway/Web/nginx-default.conf /etc/nginx/conf.d/default.conf
```
Comment (techgeek03) [inline Source/NotificationGateway/Web/Dockerfile]: You can swap these two lines to leverage layering in the image. The `cofing` file is not changing that often.

```suggestion
COPY --from=build /CBS/Source/NotificationGateway/Web/nginx-default.conf /etc/nginx/conf.d/default.conf
COPY --from=build /CBS/Source/NotificationGateway/Web/dist /usr/share/nginx/html
```

**Comment by jakhog** [comment]
Wow, didn't know that Docker was clever enough to manage the build cache on multi-stage builds, but sure enough it does (had to try it for myself). Thanks for the tip @techgeek03 👍

**Pull request merged**